src/TextField/TextField.jsx

```diff
@@ -115,6 +120,16 @@ function isValid(value) {
 const TextField = React.createClass({
 
   propTypes: {
+    /**
+     * Maximum number of character in the Textfield
+     */
+    charCountMax: React.PropTypes.number,
```

Comment:
I think `maxlength` would block the input from going over the max character count, which can be exceeded with a warning.